src/clients/api/queries/getXvsReward/index.ts

```diff
@@ -0,0 +1,107 @@
+import Web3 from 'web3';
+import BigNumber from 'bignumber.js';
+
+import { getToken } from 'utilities';
+import { VBEP_TOKENS, VBEP_TOKEN_DECIMALS } from 'constants/tokens';
+import { Comptroller } from 'types/contracts';
+import getVTokenData from './getVTokenData';
+
+export interface IGetXvsRewardInput {
+  web3: Web3;
+  accountAddress: string;
+  comptrollerContract: Comptroller;
+  venusInitialIndex: string;
+  xvsAccrued: BigNumber;
+  vaiMintIndex: string;
+  userVaiMintIndex: string;
+  userMintedVai: BigNumber;
+}
+
+export type GetXvsRewardOutput = BigNumber;
+
+// @TODO: refactor once method to fetch XVS reward from a contract has been
+// implemented (see https://app.clickup.com/t/284g9ed)
+
+const getXvsReward = async ({
+  web3,
+  accountAddress,
+  comptrollerContract,
+  venusInitialIndex,
+  xvsAccrued,
+  vaiMintIndex,
+  userVaiMintIndex,
+  userMintedVai,
+}: IGetXvsRewardInput): Promise<GetXvsRewardOutput> => {
+  const vTokensData = await Promise.all(
+    Object.values(VBEP_TOKENS).map(vToken =>
+      getVTokenData({
+        web3,
+        comptrollerContract,
+        tokenId: vToken.id,
+        tokenAddress: vToken.address,
+        accountAddress,
+      }),
+    ),
+  );
+
+  const xvsEarned = vTokensData.reduce(
```

Comment:
Similarly I wonder if we shouldn't keep queries as pure fetching and setup and transformations of the returned data in a hook